provider/internal/builtin/service_client.go

```diff
@@ -87,17 +87,28 @@ func (p *builtintServiceClient) Evaluate(cap string, conditionInfo []byte) (prov
 			return response, fmt.Errorf("could not run grep with provided pattern %+v", err)
 		}
 		matches := strings.Split(strings.TrimSpace(string(outputBytes)), "\n")
-		if len(matches) != 0 {
-			response.Matched = true
+
+		var r *regexp.Regexp
+		if c.FilePattern != "" {
+			r, err = regexp.Compile(c.FilePattern)
+			if err != nil {
+				return provider.ProviderEvaluateResponse{}, err
+			}
 		}
 
 		for _, match := range matches {
 			//TODO(fabianvf): This will not work if there is a `:` in the filename, do we care?
 			pieces := strings.SplitN(match, ":", 3)
 			if len(pieces) != 3 {
 				//TODO(fabianvf): Just log or return?
+				//(shawn-hurley): I think the return is good personally
```

Comment:
this is how discussions should happen

Reply:
I thought it would be good to add, in case someone starts to go through the todo's :)